octopus-agent/src/main/java/octopus/teamcity/agent/OctopusBuildInformationBuildProcess.java

```diff
@@ -40,146 +40,145 @@
-          overwriteMode = OverwriteMode.IgnoreIfExists;
-        }
+    @Override
+    protected String getLogMessage() {
+        return "Pushing build information to Octopus server";
+    }
 
-        if (verboseLogging) {
-          buildLogger.message("ForcePush: " + forcePush);
-          buildLogger.message("OverwriteMode: " + overwriteMode.name());
-        }
+    @Override
+    protected OctopusCommandBuilder createCommand() {
 
-        commands.add("build-information");
-        commands.add("--server");
-        commands.add(serverUrl);
-        commands.add("--apikey");
-        commands.add(masked ? "SECRET" : apiKey);
+        final BuildProgressLogger buildLogger = getLogger();
 
-        if (spaceName != null && !spaceName.isEmpty()) {
-          commands.add("--space");
-          commands.add(spaceName);
-        }
+        final Map<String, String> parameters = getContext().getRunnerParameters();
+        final OctopusConstants constants = OctopusConstants.Instance;
+        final Boolean verboseLogging =
+                Boolean.parseBoolean(parameters.get(constants.getVerboseLoggingKey()));
 
-        for (String packageId : StringUtil.split(packageIds, "\n")) {
-          commands.add("--package-id");
-          commands.add(packageId);
-        }
+        final String dataFile =
+                Paths.get(checkoutDir.getPath(), "octopus.buildinfo").toAbsolutePath().toString();
 
-        commands.add("--version");
-        commands.add(packageVersion);
+        try {
+            AgentRunningBuild build = getContext().getBuild();
 
-        commands.add("--file");
-        commands.add(dataFile);
+            final OctopusBuildInformationBuilder builder = new OctopusBuildInformationBuilder();
 
-        if (overwriteMode != OverwriteMode.FailIfExists) {
-          commands.add("--overwrite-mode");
-          commands.add(overwriteMode.name());
-        }
+            final String buildIdString = Long.toString(build.getBuildId());
+            final TeamCityInstance teamCityServer =
+                    TeamCityInstanceFactory.httpAuth(
+                            teamCityServerUrl, build.getAccessUser(), build.getAccessCode());
+            final Build restfulBuild = teamCityServer.build(new BuildId(buildIdString));
 
-        if (commandLineArguments != null && !commandLineArguments.isEmpty()) {
-          commands.addAll(splitSpaceSeparatedValues(commandLineArguments));
+            final OctopusBuildInformation buildInformation =
+                    builder.build(
+                            sharedConfigParameters.get("octopus_vcstype"),
+                            sharedConfigParameters.get("vcsroot.url"),
+                            sharedConfigParameters.get("build.vcs.number"),
+                            restfulBuild.getBranch().getName(),
+                            createJsonCommitHistory(restfulBuild),
+                            sharedConfigParameters.get("externalBuildURL"),
```

Comment:
This wall of text bought to you by an IDE reformat of the code to correct it's layout. This is the key line that actually changed.